File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/exception/CapacityExceededException.java
package com.dominikgaller.alpinebooking.booking.core.domain.exception;

/**
 * Thrown when the requested participant count exceeds the available capacity for a tour.
 *
 * <p>Maps to HTTP 409 Conflict at the REST boundary.
 *
 * <p>SDD: See {@code documentation/domain/aggregate-tour-booking.spec.md}, section 2.
 */
public class CapacityExceededException extends RuntimeException {

    public CapacityExceededException(final int requested, final int available) {
        super(String.format(
                "Requested %d participants exceeds available capacity of %d",
                requested, available));
    }
}

File: src/main/kotlin/com/example/contractmanagement/booking/core/domain/tourbooking/exception/CapacityExceededException.kt
package com.example.contractmanagement.booking.core.domain.tourbooking.exception

/**
 * Thrown when the requested participant count exceeds the capacity available at the time
 * of the request.
 *
 * Maps to HTTP 409 (`coding-style.definition.md` § 6.2): the request is well-formed, the
 * world simply does not allow it right now.
 */
class CapacityExceededException(
    requested: Int,
    available: Int,
) : RuntimeException("Requested $requested participants exceeds available capacity of $available")
